### PR TITLE
fix: maintain dependabot config filename for existing configs

### PR DESCRIPTION
**Pull request by zkoppert**
# Pull Request

## Proposed Changes

In order for us to handle existing configuration files properly, we need to ensure we are making a pull request against the existing configuration and not a new file. This is occurring currently when the file dependabot.yml has a slightly different extension (.yaml).

### Minor Changes:

**Updates to GitHub Token and Commit Message:**

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L59-R59): The `GH_TOKEN` description is updated to specify that it must also have `repo:write` and `workflow` privileges to create a pull request.
* [`env.py`](diffhunk://#diff-13072a46fe571952a394d1f742b8056193bc2c30256480e7066ff07438cc3764L190-R190): The default commit message is changed from "Create dependabot.yaml" to "Create/Update dependabot.yaml".

### Most significant changes:

**Changes in `dependabot_file.py`:**

* The function `make_dependabot_config` now takes an additional `indent` parameter to control the indentation of the generated `dependabot.yaml` content.
* The `build_dependabot_file` function has been updated to calculate the indentation based on the existing `dependabot.yaml` file, if it exists.
* The `build_dependabot_file` function now passes the calculated `indent` to `make_dependabot_config` when adding new package managers to the `dependabot.yaml` file. [[1]](diffhunk://#diff-9c1e512a78997d7d924506ea0d4abe83e23c5d35c7df780a0ec6406fd3e3650eL102-R113) [[2]](diffhunk://#diff-9c1e512a78997d7d924506ea0d4abe83e23c5d35c7df780a0ec6406fd3e3650eL115-R126) [[3]](diffhunk://#diff-9c1e512a78997d7d924506ea0d4abe83e23c5d35c7df780a0ec6406fd3e3650eL126-R137)

**Changes in `evergreen.py`:**

* The `main` function now keeps track of the filename of the existing `dependabot.yaml` file, if it exists.
* The filename of the `dependabot.yaml` file is now included in the issue body when creating an issue.
* The `commit_changes` function now takes additional parameters for the filename of the `dependabot.yaml` file and the existing configuration. If the configuration exists, it updates the file; otherwise, it creates a new one. [[1]](diffhunk://#diff-b2e641407994f0e45d80e45129e5d0388cc50cdfe231fe65d25603aa7c742293L154-R164) [[2]](diffhunk://#diff-b2e641407994f0e45d80e45129e5d0388cc50cdfe231fe65d25603aa7c742293L276-R310)

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing
- [x] If publishing new data to the public (scorecards, security scan results, code quality results, live dashboards, etc.), please request review from `@jeffrey-luszcz`

### Reviewer

- [x] Label as either `fix`, `documentation`, `enhancement`, `infrastructure`, `maintenance` or `breaking`

### CODE REVIEW

File: evergreen.py [hunk 1]
```diff
@@ -282,7 +293,7 @@ def commit_changes(title, body, repo, dependabot_file, message):
     branch_name = "dependabot-" + str(uuid.uuid4())
     repo.create_ref(front_matter + branch_name, default_branch_commit)
     repo.create_file(
-        path=".github/dependabot.yaml",
+        path=".github/" + dependabot_filename,
```
Comment (zkoppert) [inline evergreen.py:296]: found a bug here where it adds ".github/ twice. fixing locally and will push new commit to fix.

File: test_evergreen.py [hunk 2]
```diff
@@ -205,20 +205,28 @@ def test_commit_changes(self, mock_uuid):
...
+            dependabot_file_name,
+        )
 
         # Assert that the methods were called with the correct arguments
         mock_repo.create_ref.assert_called_once_with(
             f"refs/heads/{branch_name}", "abc123"
         )
         mock_repo.create_file.assert_called_once_with(
-            path=".github/dependabot.yaml",
+            path=".github/" + dependabot_file_name,
```
Comment (jmeridth) [inline test_evergreen.py:229]: you'll need to fix here too.

**Review by jmeridth — APPROVED** [review]
looks good.  gonna approve since you're already gonna fix the filepath issue.

**Comment by zkoppert** [comment]
@jmeridth I ended up down a bit of a rabbit hole with this one. I re-requested your review because I changed more than I advertised.

File: README.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ This action can be configured to authenticate with GitHub App Installation or Pe
 
 | field                         | required | default | description |
 |-------------------------------|----------|---------|-------------|
-| `GH_TOKEN`                    | True     | `""`    | The GitHub Token used to scan the repository. Must have read access to all repository you are interested in scanning. |
+| `GH_TOKEN`                    | True     | `""`    | The GitHub Token used to scan the repository. Must have read access to all repository you are interested in scanning and repo:write and workflow privelages to create a pull request. |
```
Comment (jmeridth) [inline README.md:59]: Did we mean to use `...and repo:write and...`.  Am I misreading the second `and`?

File: dependabot_file.py [hunk 4]
```diff
@@ -61,7 +62,17 @@ def build_dependabot_file(
...
+            line for line in dependabot_file.splitlines() if "directory:" in line
+        )
+        indent = " " * (len(directory_line) - len(directory_line.lstrip()))
+        if len(indent) < 2:
+            print(
+                "Invalid dependabot.yml file. No indentation found. Skipping {repo.full_name}"
+            )
+            return None
     else:
+        indent = " " * 2
```
Comment (jmeridth) [inline dependabot_file.py:75]: could we move the `2` to the top of the file and call it something like `DEFAULT_INDENT = 2`? wdyt?

File: dependabot_file.py [hunk 5]
```diff
@@ -61,7 +62,17 @@ def build_dependabot_file(
 
     if existing_config:
         dependabot_file = existing_config.decoded.decode("utf-8")
+        directory_line = next(
+            line for line in dependabot_file.splitlines() if "directory:" in line
+        )
+        indent = " " * (len(directory_line) - len(directory_line.lstrip()))
+        if len(indent) < 2:
```
Comment (jmeridth) [inline dependabot_file.py:69]: can check the new `DEFAULT_INDENT` here also, once created

File: test_dependabot_file.py [hunk 6]
```diff
@@ -54,7 +54,37 @@ def test_build_dependabot_file_with_existing_config_bundler_no_update(self):
...
+    ):
+        """Test that the dependabot.yml file is built correctly with bundler"""
+        repo = MagicMock()
+        repo.file_contents.side_effect = lambda f, filename="Gemfile": f == filename
+
+        # expected_result maintains existing ecosystem with custom configuration and adds new ecosystem
+        expected_result = """---
+version: 2
+updates:
+- package-ecosystem: "pip"
```
Comment (jmeridth) [inline test_dependabot_file.py:68]: maybe it is just the diff view, but shouldn't there be 2 spaces here on each line?  Same with the rest of the tests.  Default indent is 2, correct?

**Review by jmeridth — COMMENTED** [review]
Couple questions/suggestions

File: README.md (hunk 3 above)
Comment (zkoppert) [reply README.md:59]: Is this more clear?
```suggestion
| `GH_TOKEN`                    | True     | `""`    | The GitHub Token used to scan the repository. Must have read access to all repository you are interested in scanning, `repo:write`, and `workflow` privileges to create a pull request. |
```

File: test_dependabot_file.py (hunk 6 above)
Comment (zkoppert) [reply test_dependabot_file.py:68]: The "- " replaces the last two chars of the indent as shown in these docs. Code for that is here (notice the indent[:-2]) :
```python
dependabot_config = f"""{indent[:-2]}- package-ecosystem: '{ecosystem}'
{indent}directory: '/'
{indent}schedule:
{indent}{indent}interval: 'weekly'
```

That being said, I'm not sure this is the correct thing to do. 😮‍💨 

[The docs](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#example-of-a-basic-setup-for-three-package-managers) demonstrate one spacing scheme:
```yml
version: 2
updates:
  - package-ecosystem: "pip"
    directory: "/"
```

while internally at github (even in the monolith) I've also seen:
```yml
version: 2
updates:
- package-ecosystem: "pip"
  directory: "/"
```

Comment (zkoppert) [reply test_dependabot_file.py:68]: I'm inclined to go with the docs on this one and fix our internal "misuse" of the odd alignment.

File: dependabot_file.py [hunk 7]
```diff
@@ -58,10 +59,23 @@ def build_dependabot_file(
...
+        ecosystem_line = next(
+            line
+            for line in dependabot_file.splitlines()
+            if "- package-ecosystem:" in line
+        )
+        indent = " " * (len(ecosystem_line) - len(ecosystem_line.lstrip()))
+        if len(indent) < DEFAULT_INDENT:
+            print(
+                "Invalid dependabot.yml file. No indentation found. Skipping {repo.full_name}"
+            )
```
Comment (zkoppert) [inline dependabot_file.py:75]: This should be an f-string :doh:
```suggestion
            print(
                f"Invalid dependabot.yml file. No indentation found. Skipping {repo.full_name}"
            )
```

**Pull request merged**